opentelemetry-sdk/src/trace/mod.rs

```diff
@@ -76,6 +76,49 @@ mod tests {
         Context, KeyValue,
     };
 
+    #[test]
+    fn span_context_immutable() {
```

Comment:
https://github.com/open-telemetry/opentelemetry-rust/pull/1268 is where we stored Span separate from the HashMap as an Arc. @shaun-cox can you share your understanding? Is it intentional that Span was Arc wrapped?

Reply:
@cijothomas that change does not change anything with regards to the `Span` being mutable. Before it was stored in the `HashMap<TypeId, Arc<dyn Any + Sync + Send>, BuildHasherDefault<IdHasher>>`, still an `Arc` pointing to the same `SynchronizedSpan`, and then it was moved to a separate field for performance reasons.

Reply:
oh yes, that is correct!